### PR TITLE
Check by 

### PR DESCRIPTION
**Pull request by searchformyusername**
@Geek-ubaid

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,47 +1,59 @@
...
+[![DOCS](https://img.shields.io/badge/HelpersGuide-see%20help-green?style=flat-square&logo=appveyor)](https://github.com/D-E-F-E-A-T/Selenium-Cookie-Injector/blob/master/helpers.md)
 
 <br>
 
-## Contributors
+## Functionalities
+- [x] Cookie adding from other Browsers cookie
+- [ ] Explicit and Selected Cookie Addition
+- [ ] Passwords Addition from other Browsers to Selenium Browser
+
```
Comment (Geek-ubaid) [inline README.md:60]: Move the functionalities under project description. Don't change the order.

File: SeleniumCookie/wrapper.py [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+
+    from pysqlite2 import dbapi2 as sqlite3
+except ImportError:
+    import sqlite3
+
+# external dependencies
+import keyring
+import pyaes
+from pbkdf2 import PBKDF2
+
```
Comment (Geek-ubaid) [inline SeleniumCookie/wrapper.py:27]: Group all the default packages like sys is etc together and all the third party packages together. This is a good practice!

File: helpers.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+		pass
+```
+</br>
+
+## Testing
+* Run the above script
+* Now in Chrome Search Bar: chrome://settings/siteData
+* Added Cookies can be seen now
+* Additon of cookie subject to it's various parameters and data it has
+* In my test 279 (119 websites) cookies were injected and 2924 were rejected
```
Comment (Geek-ubaid) [inline helpers.md]: For testing right down unittest using python. Also the things mentioned here come under results. For unittest refer this https://realpython.com/python-testing/

File: setup.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
+from distutils.core import setup
+setup(
+  name = 'SeleniumCookie',
+  packages = ['SeleniumCookie'],
+  version = '0.1',
+  license='MIT',
+  description = 'COOKIE INJECTION FOR SELENIUM',
+  author = 'D-E-F-E-A-T, L04DB4L4NC3R',
```
Comment (Geek-ubaid) [inline setup.py]: Multiple author name comes in list. Just check this out

**Review by Geek-ubaid — CHANGES_REQUESTED** [review]
Some changes to be made :

1. Add all the import function at one place i.e. above the code

2. Define the docstring for all global functions

3. Follow the pip directory structure. Include test folder in the repo which should include unittest for the code.

4. Correct the spelling of license

**Comment by L04DB4L4NC3R** [comment]
@D-E-F-E-A-T Give proper name of the pull request. See pull request formats online

**Comment by L04DB4L4NC3R** [comment]
Write informative commit messages and avoid repetitive commits

image

File: helpers.md (hunk 3 above)
Comment (L04DB4L4NC3R) [reply helpers.md]: Instead of helpers.md, just add the installation instructions(or anything else) in the base README

File: README.md [hunk 5]
```diff
@@ -51,10 +51,10 @@ for cookie in cookies:
 ## Contributors
 
 * [ Angad ](https://github.com/L04DB4L4NC3R)
+* [ Ubaid ](https://github.com/Geek-ubaid/)
```
Comment (L04DB4L4NC3R) [inline README.md]: Only add names of the developers, not maintainers

File: SeleniumCookie/wrapper.py [hunk 6]
```diff
@@ -0,0 +1,353 @@
...
+import lz4.block
+import datetime
+import configparser
+
+try:
+    import json
+except ImportError:
+    import simplejson as json
+try:
+
```
Comment (L04DB4L4NC3R) [inline SeleniumCookie/wrapper.py]: Remove space

File: SeleniumCookie/wrapper.py [hunk 7]
```diff
@@ -0,0 +1,353 @@
...
+class BrowserCookieError(Exception):
+    pass
+
+
+def create_local_copy(cookie_file):
+
+    if os.path.exists(cookie_file):
+
+        tmp_cookie_file = tempfile.NamedTemporaryFile(suffix='.sqlite').name
+        open(tmp_cookie_file, 'wb').write(open(cookie_file, 'rb').read())
```
Comment (L04DB4L4NC3R) [inline SeleniumCookie/wrapper.py]: Break this action down to multiple steps for better understandability

File: SeleniumCookie/wrapper.py [hunk 8]
```diff
@@ -0,0 +1,353 @@
...
+        return tmp_cookie_file
+    else:
+        raise BrowserCookieError('Can not find cookie file at: ' + cookie_file)
```
Comment (L04DB4L4NC3R) [inline SeleniumCookie/wrapper.py]: Use formatting instead of string concatenation

**Review by L04DB4L4NC3R — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (searchformyusername) [reply README.md:60]: These are the feature ideas that can be added in future. first one is ticked which is added.

File: README.md (hunk 5 above)
Comment (searchformyusername) [reply README.md]: ok but i got help from u 2

File: helpers.md (hunk 3 above)
Comment (D-E-F-E-A-T) [reply helpers.md]: Please let it be helpers guide. that feels more appropriate.

Comment (D-E-F-E-A-T) [reply helpers.md]: @Geek-ubaid thanks for this small blog (*small*)

**Review by L04DB4L4NC3R — APPROVED** [review]

**Review by Geek-ubaid — APPROVED** [review]

**Pull request merged**